### DISCUSSION TITLE
Errors reading EXR images with Houdini 19.5 on Windows.

### DISCUSSION THREAD

**Discussion by ahemberger** [colour-science/colour#1019] [Q&A]
I seem to repeatedly run up against this problem every few times I upgrade Houdini. I'm installing colour into Houdini 19.5 (which ships with Python 3.9), and am trying to use it to read an image:

`img_data = colour.read_image('image.exr')`

freeimage throws this error:
```
ValueError: Could not load bitmap "image.exr": Cannot read image file "". The file format version number's flag field contains unrecognized flags.
```

Any idea how to troubleshoot this? I understand the real issue here is with freeimage and not colour. I run into this type of problem (freeimage not handling exrs well) that I wonder if using a different module to handle i/o might be less of a headache...

Thanks!

**Comment by thomasmansencal** [comment]
Hi @ahemberger,

Colour will use "Imageio" by default and will prefer "OpenImageIO" if it is available. Having just tried on my MacBook with a fresh 19.5 Apprentice:

```python
>>> import imageio
>>> imageio.imread("/Users/kelsolaar/Downloads/CORRECTED.exr")
Array([[[0.83984375, 0.9038086 , 0.93115234],
        [0.8417969 , 0.90478516, 0.9321289 ],
        [0.8432617 , 0.9057617 , 0.9326172 ],
        ...,
```

"OpenImageIO" is not shipped that said:

```python
>>> import OpenImageIO
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Applications/Houdini/Houdini19.5.303/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-pac
kages-forced/shiboken2/files.dir/shibokensupport/__feature__.py", line 142, in _import
    return original_import(name, *args, **kwargs)
ModuleNotFoundError: No module named 'OpenImageIO'
```

**Reply by ahemberger** [reply]
Hi Thomas;

Thanks, yes, I understand as much. It's my understanding that imageio uses freeimage to read EXRs, and it is in Freeimage that the above error is written (well, specifically ImfHeader.cpp). You advised me as much here:
https://github.com/colour-science/colour/discussions/984

The image I'm trying to read was _written_ by H19.5, and the code I'm using worked fine in H19, so I suspect something may have changed in how Houdini is constructing the exr file. After a bit of experimenting, I found a parameter on the USD Render node in Houdini (sidefx rewrote this a bit for H19.5):

```
Enable Legacy EXR Mode

When saving OpenEXR images with multiple AOVs, husk will normally save the images using multi-part EXR files. This extension (introduced in 2013) lets programs read a single AOV without having to read the entire image. This allows most software to work much more efficiently.

Some applications have not transitioned to support multi-part images. If your pipeline uses software that doesn’t support OpenEXR v2.0, this option will force husk to write out images in an inefficient, but more compatible format.
```
When I enable this option, freeimage seems to make sense of my file fine. This makes me suspect that prior versions of Houdini weren't writing out multi-part EXRs by default, and that perhaps freeimage/imageio don't support it yet. 

Anyway, leaving this here as a breadcrumb in case anyone else runs into this issue.
